chapters/05-version-control-solo.Rmd

```diff
@@ -12,6 +12,115 @@ Clearly identify which tasks are best accomplished by each interface.
+* publish repositories to GitHub and coordinate between remote and local versions
+
+## Overview
+
+All text (no code):
+* why to use version control
+* why to use Git as a version control system (popularity, GitHub, 
+* how are Git and GitHub different?
+  * local vs remote
+  * publishing 
+* why use version control for software engineering 
+  * basics of version control and why it matters for code
+  * preview this chapter, connect to other chapters
+
+## Git integration in RStudio
+
+### Setting up Git in RStudio
+
+* text: connecting Git and RStudio
+  * accessing Git through Global options
+  * tab in environment pane 
+* code: `usethis::use_git` to initialize repository
+* text: explain what happens when repo is initialized
+  * creates hidden folder containing git history
+  * some files ignored (note that it's possible to add other files to ignore)
+
+### Tracking changes in RStudio
+
+All text and images:
+* exercise: work through basic workflow using RStudio integration
+  * all existing files in repo appear in git pane
+  * ensure box is checked so these files are staged
+  * click Commit button, add commit message and click commit
+  * note pop up box with commit report, files no longer appear in window
+* reiterate workflow by editing README
+  * status difference between tracking new files and editing previously tracked file
+
+* exercise: create new file, commit, delete file, commit
+
+### Viewing history in RStudio
+
+All text and images:
+* click on clock button in Git pane (don't confuse with history tab; all git buttons are in Git tab)
+* explain fields: subject, author, date, SHA
+* click on a previous commit to see changes
+
+## Git in Terminal
+
+* text: git on command line includes commands for full range of tasks, including those we've already covered:
+  * initialize repository: `git init`
+  * basic workflow: `git add` then `git commit`
+  * view history: `git log`
+
+### Exploring history in Terminal
+
+* code: make a change to README, then restore to previous (most recent) commit (pretend changes didn't exist, as if we changed our mind)
+  * `git status`
+  * `git restore README.md`
+  * `git status`
+
+* text: if we wanted to use content from a previous commit, like the file we created then deleted earlier
+  * find SHA key in log
+* code:
+  * `git checkout XXXXXXX filename`
+  * `git status`
+  * `git restore --staged filename`
+  * `git restore filename`
+  * `git status`
+
+### Creating branches in Terminal
+
+* text: what is a branch? 
+  * main branch (make sure usethis creates main as default branch)
+  * why work with them?
+* code: create branch, make changes, merge with main
+  * `git branch`
+  * `git checkout -b new_idea`
+  * make changes to file
+  * `git add filename`
+  * `git commit -m filename`
+  * `git checkout main`
+  * `git merge new_idea`
+  * `git branch -d new_idea`
+* text: note working with branches in rstudio tab
```

Comment:
Here I would also maybe say that we will cover using branches with GitHub in later chapter?

Reply:
I feel like this chapter should introduce using branches, since that's a reasonable part of a standard git process for someone working alone. Can I move a few things from the git workflow chapter here? I'm thinking specifically of deleting a branch that isn't going to be used.